baseframe/forms/validators.py

```diff
@@ -16,42 +17,79 @@
 from ..signals import exception_catchall
 
 
-__all__ = ['OptionalIf', 'OptionalIfNot', 'ValidEmail', 'ValidEmailDomain', 'ValidUrl', 'AllUrlsValid',
-    'ValidName', 'NoObfuscatedEmail', 'ValidCoordinates', 'IsPublicEmailDomain', 'IsNotPublicEmailDomain',
-    # WTForms validators
-    'DataRequired', 'InputRequired', 'Optional', 'Length', 'EqualTo', 'URL', 'NumberRange',
-    'ValidationError', 'StopValidation']
+__local = ['AllUrlsValid', 'IsNotPublicEmailDomain', 'IsPublicEmailDomain', 'NoObfuscatedEmail',
+    'AllowedIf', 'OptionalIf', 'RequiredIf', 'ValidCoordinates', 'ValidEmail',
+    'ValidEmailDomain', 'ValidName', 'ValidUrl']
+__imported = [  # WTForms validators
+    'DataRequired', 'EqualTo', 'InputRequired', 'Length', 'NumberRange', 'Optional',
+    'StopValidation', 'URL', 'ValidationError']
+__all__ = __local + __imported
 
 
 EMAIL_RE = re.compile(r'\b[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,63}\b', re.I)
 
 
-class OptionalIf(object):
+class AllowedIf(object):
     """
-    Validator that makes this field optional if the value of some other field is true.
+    Validator that allows a value only if another field also has a value.
+
+    :param str fieldname: Name of the other field
+    :param str message: Validation error message. Will be formatted with an optional ``{field}}`` label
+    """
+    def __init__(self, fieldname, message=None):
+        self.fieldname = fieldname
+        self.message = message or __(u"This requires {field} to be specified")
+
+    def __call__(self, form, field):
+        if field.data:
+            if form[self.fieldname].data in (None, ''):
+                raise StopValidation(self.message.format(field=form[self.fieldname].label.text))
+
+
+class OptionalIf(Optional):
+    """
+    Validator that makes this field optional if another field has data. If this
+    field is required when the other field is empty, chain it with
+    :class:`DataRequired`::
```

Comment:
Doesn't `OptionalIf` name imply that it'll be Required otherwise? Same question with `RequiredIf`